src/magic_enchantment.cpp

```diff
@@ -631,10 +719,30 @@ void enchantment::cast_enchantment_spell( Character &caster, const Creature *tar
     }
 }
 
-bool enchantment::operator==( const enchantment &rhs ) const
+bool enchant_cache::operator==( const enchant_cache &rhs ) const
 {
+    if( this->values_add.size() != rhs.values_add.size() ||
+        this->values_multiply.size() != rhs.values_multiply.size() ) {
+        return false;
+    }
+    auto iter_add = this->values_add.cbegin();
+    auto iter_add2 = rhs.values_add.cbegin();
+    while( iter_add != this->values_add.cend() && iter_add2 != rhs.values_add.cend() ) {
+        if( iter_add->second != iter_add2->second ) {
+            return false;
+        }
+        iter_add++;
+        iter_add2++;
+    }
+    auto iter_mult = this->values_multiply.cbegin();
+    auto iter_mult2 = rhs.values_multiply.cbegin();
+    while( iter_mult != this->values_multiply.cend() && iter_mult2 != rhs.values_multiply.cend() ) {
```

Comment:
Another thing I noticed here: is it correct to compare only the *values* of the map, but not the keys? Currently a map [a => 1, b => 2] is equal to a map [x => 1, y => 2] because the values (1, 2) are equal and in same order, but the maps have completely different keys.